include/odp/api/spec/proto_stats.h

```diff
@@ -0,0 +1,205 @@
+} odp_proto_stats_param_t;
+
+/**
+ * Proto stats capabilities
+ */
+typedef struct odp_proto_stats_capability_t {
+	/** Supported proto stats id mask
+	 *
+	 * A bit is set at position ODP_PROTO_STATS_ID_*,
+	 * when that proto stats id is supported.
+	 */
+	uint64_t stats_mask_supported;
+
+	/** ODP_PROTO_STATS_ID_TX_* flags */
+	struct {
+		/** Packet adjust support for ODP_PROTO_STATS_ID_TX_OCT_COUNT0 */
+		odp_bool_t oct_count0_adj;
+
+		/** Packet adjust support for ODP_PROTO_STATS_ID_TX_OCT_COUNT1 */
+		odp_bool_t oct_count1_adj;
+	} tx;
+} odp_proto_stats_capability_t;
+
+/**
+ * Initialize proto stats parameters
+ *
+ * Initialize an odp_proto_stats_param_t to its default values.
+ * By default all the statistics are disabled.
+ *
+ * @param param   Proto stats parameter pointer.
+ */
+void odp_proto_stats_param_init(odp_proto_stats_param_t *param);
+
+/**
+ * Get proto stats capability
+ *
+ * Get supported protocol statistics and metadata.
+ *
+ * @param capa Pointer where capabilities are updated.
+ */
+void odp_proto_stats_capability(odp_proto_stats_capability_t *capa);
+
+/**
+ * Create a proto stats object
+ *
+ * Create a proto stats object with given name and parameters.
+ *
+ * @param name  Object name
+ * @param param Proto stats parameters
+ *
+ * @return Proto stats object handle
+ * @retval ODP_PROTO_STATS_INVALID on failure
+ */
+odp_proto_stats_t odp_proto_stats_create(const char *name, const odp_proto_stats_param_t *param);
+
+/**
+ * Lookup a proto stats object by name
+ *
+ * Lookup an already created proto stats object by name.
+ *
+ * @param name Proto stats object name
+ *
+ * @return Proto stats object handle
+ * @retval ODP_PROTO_STATS_INVALID on failure
+ */
+odp_proto_stats_t odp_proto_stats_lookup(const char *name);
+
+/**
+ * Destroy a proto stats object
+ *
+ * Destroy a proto stats object already created.
+ *
+ * - Before destroying proto stats object having ODP_PROTO_STATS_ID_TX_*
+ *   enabled, for all PKTIO devices to which packets were Tx'ed earlier with
+ *   this proto stats object, odp_pktio_stop() must be called. Additionally,
+ *   existing packets that refer to the proto stats object being destroyed
+ *   must not be sent at the same time as or after the proto stats object
+ *   destruction.
+ *
+ * @param stat Proto stats handle
+ *
+ * @retval 0 on success
+ * @retval <0 on failure
+ */
+int odp_proto_stats_destroy(odp_proto_stats_t stat);
+
+/**
+ * Get proto stats counter by id
+ *
+ * Get current value of statistics counter by stat id.
+ *
+ * @param stat Proto stat object handle
+ * @param id   Stats counter identifier to return the value.
+ *
+ * @return Stats counter value
+ * @retval Unknown value if stats id is not valid for the object.
+ */
+uint64_t odp_proto_stats(odp_proto_stats_t stat, odp_proto_stats_id_t id);
+
+/** ODP proto stats counters */
+typedef struct odp_proto_stats_data_t {
+	/** Array of counters */
+	uint64_t counter[ODP_PROTO_STATS_ID_MAX];
+} odp_proto_stats_data_t;
```

Comment:
In order to match with other stats APIs, this struct could list each counter by its name. Application would not need to use counter ID enum for reading counters. Currently, enum fixes the order of the counters anyway.